### ISSUE TITLE
Dynamodb returns 500 for TransactWriteItems with ConditionCheck without hit items

### ISSUE THREAD

**Issue by highmtworks** [localstack/localstack#3801]
# Type of request: This is a ...

- [x] bug report
- [ ] feature request

# Detailed description

dynamodb transact-write-items fails with 500 if:
* Update with ConditionExpression
* and ConditionExpression fails,
* and the item of the specified key does not exist

This is not a duplicate of #2745 (maybe by the third condition above.)

See _Steps to reproduce_ section for more details.

...

## Expected behavior

The operation should fail with ConditionalCheckFailed like:

```
An error occurred (TransactionCanceledException) when calling the TransactWriteItems operation: Transaction cancelled, please refer cancellation reasons for specific reasons [ConditionalCheckFailed]
```

## Actual behavior

The operation fails with 500 like:

```
An error occurred (500) when calling the TransactWriteItems operation (reached max retries: 2):
```

and localstack outputs error logs:
```
WARNING:bootstrap.py: Thread run method <function AdaptiveThreadPool.submit.<locals>._run at 0x7ff7a56591f0>(None) failed: 'NoneType' object is not iterable Traceback (most recent call last):
  File "/opt/code/localstack/localstack/utils/bootstrap.py", line 672, in run
    result = self.func(self.params)
  File "/opt/code/localstack/localstack/utils/async_utils.py", line 28, in _run
    return fn(*args, **kwargs)
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 607, in handler
    response = modify_and_forward(method=method, path=path_with_params, data_bytes=data, headers=headers,
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 373, in modify_and_forward
    listener_result = listener.forward_request(method=method,
  File "/opt/code/localstack/localstack/services/edge.py", line 108, in forward_request
    return do_forward_request(api, method, path, data, headers, port=port)
  File "/opt/code/localstack/localstack/services/edge.py", line 119, in do_forward_request
    result = do_forward_request_inmem(api, method, path, data, headers, port=port)
  File "/opt/code/localstack/localstack/services/edge.py", line 139, in do_forward_request_inmem
    response = modify_and_forward(method=method, path=path, data_bytes=data, headers=headers,
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 443, in modify_and_forward
    updated_response = update_listener.return_response(**kwargs)
  File "/opt/code/localstack/localstack/services/dynamodb/dynamodb_listener.py", line 295, in return_response
    for record in records:
TypeError: 'NoneType' object is not iterable
```

# Steps to reproduce

## Command used to start LocalStack

```
docker run -it --rm -p4566:4566 localstack/localstack:0.12.9
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

```
# aws --version
# aws-cli/2.1.32 Python/3.8.8 Windows/10 exe/AMD64 prompt/off
set AWS_ACCESS_KEY_ID=localstack
set AWS_SECRET_ACCESS_KEY=localstack
aws --endpoint http://localhost:4566 dynamodb create-table --table-name testtab  --key-schema "[{\"AttributeName\":\"testid\",\"KeyType\":\"HASH\"}]" --attribute-definitions "[{\"AttributeName\":\"testid\",\"AttributeType\":\"S\"}]" --billing-mode PAY_PER_REQUEST
aws --endpoint http://localhost:4566 dynamodb transact-write-items --transact-items "{\"Update\":{\"TableName\":\"testtab\",\"Key\":{\"testid\":{\"S\":\"id1\"}},\"UpdateExpression\":\"SET updated = :updated\",\"ConditionExpression\":\"attribute_exists(testid)\",\"ExpressionAttributeValues\":{\":updated\":{\"BOOL\":true}}}}"
```

----
As a workaround,
modifying by:
patch/0001.patch
```
diff --git a/localstack/services/dynamodb/dynamodb_listener.py b/localstack/services/dynamodb/dynamodb_listener.py
index 75dd5e31..0c06fe19 100644
--- a/localstack/services/dynamodb/dynamodb_listener.py
+++ b/localstack/services/dynamodb/dynamodb_listener.py
@@ -468,7 +468,7 @@ class ProxyListenerDynamoDB(ProxyListener):
                     return keys
                 updated_item = find_existing_item(update_request, table_name)
                 if not updated_item:
-                    return
+                    return []
                 new_record = clone(record)
                 new_record['eventName'] = 'MODIFY'
                 new_record['dynamodb']['Keys'] = keys
```
with Dockerfile
```
FROM localstack/localstack:0.12.9

COPY patch/* /opt/code/localstack/patch/
RUN find /opt/code/localstack/patch/ -type f -exec patch -p1 -i \{\} \;

ENTRYPOINT ["docker-entrypoint.sh"]
```
and then running patched localstack
```
docker build -t localhost/my/localstack:0.12.9 .
docker run -it --rm -p4566:4566 localhost/my/localstack:0.12.9
```
produces the expected behavior,
but I could not verify whether it is correct. (e.g. behavior for ReturnValuesOnConditionCheckFailure)

**Comment by whummer** [comment]
Thanks for reporting @highmtworks . This should be fixed in #3812 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists. Thanks!

**Comment by highmtworks** [comment]
Thank you for the fix!
